### DISCUSSION TITLE
RDS Proxy having "no benefit for connection pooling..."

### DISCUSSION THREAD

**Discussion by qmitchell-aa** [prisma/prisma#23547] [Q&A] [answered]
### Question

Nov 2023, RDS Proxy started supporting protocol-level prepared statements. [link](https://aws.amazon.com/blogs/database/amazon-rds-proxy-multiplexing-support-for-postgresql-extended-query-protocol/)

Prisma documentation says that RDS Proxy has "no benefit for connection pooling..." [link](https://www.prisma.io/docs/orm/prisma-client/deployment/caveats-when-deploying-to-aws-platforms#aws-rds-proxy).  

I am wondering if this prisma documentation is out of date , and not reflective of this new RDS proxy feature.

**Comment by janpio** [comment, marked as answer]
A related issue comment from earlier this year: https://github.com/prisma/prisma/issues/5866#issuecomment-1887963763

My understanding is still that the way Prisma runs queries is unfortunately still not supported by AWS RDS Proxy (with us setting the search_path for example for each connection).

**Reply by qmitchell-aa** [reply]
Thank you, greatly appreciate the note about `search_path`.  I'll be testing shortly for my application and see how often pinning logs show up.

**Reply by qmitchell-aa** [reply]
Based on this documentation, I am thinking to set `pgbouncer mode` for when I use RDS proxy.  This would disable `search_path` as you mentioned.

<img width="710" alt="image" src="https://github.com/prisma/prisma/assets/139587172/084208e5-5eba-43c1-a6a4-e13ee711cb78">

**Reply by joshsny** [reply]
> Based on this documentation, I am thinking to set `pgbouncer mode` for when I use RDS proxy.  This would disable `search_path` as you mentioned.
> 
> 
> 
> <img width="710" alt="image" src="https://github.com/prisma/prisma/assets/139587172/084208e5-5eba-43c1-a6a4-e13ee711cb78">
> 
> 

Did this work? Interested in doing something similar.

**Reply by qmitchell-aa** [reply]
@joshsny - It's working fine for me, but it might only b/c my query volume is too low & my queries are too quick for metrics to really pop. 

My RDS is a `db.m6g.large` and that gives me ~800 DB connections of capacity, and then RDS Proxy ~400 DB connection capacity.  That's plenty of connections to do "wrong" things.  My pinned connection count definitely not zero and quite related to the client connections.  The pinned logs tell me that the `search_path` variable is set, even though I have the pgbouncer flag set.  😅 tell me how that makes sense.

**Reply by joshsny** [reply]
Thanks for the reply, that sounds interesting!

Are there any signs in CloudWatch that the number of active connections from your application to RDS Proxy is less than those from RDS Proxy to your DB, even if you are not close to the connection limits of 400/800?

**Reply by qmitchell-aa** [reply]
yeah, metrics show that the connection count from Proxy to DB is always higher than my connection count from app to Proxy. 

I feel like we'd want the reverse. ( eg Proxy to have all the connections to it directly, and only some connections to DB )

**Reply by joshsny** [reply]
Ah I guess the proxy might be keeping some connections open as a buffer, but sounds like it's not easy to determine if it's helping you or not whilst your app's connection count is so low.

I might give this a try myself and see what the impact is!

**Reply by qmitchell-aa** [reply]
👍 come back and tell me how it goes.

**Reply by tmm1** [reply]
I don't recommend setting `?pgbouncer=true` w/ RDS proxy: https://github.com/prisma/prisma-engines/pull/5016#issuecomment-2403637654

I'm not seeing any connection-pinning issues due to search_path when using the client without that flag set. Can you share the log message?
